indico/migrations/versions/20240314_1430_4e32f4d5ebe4_add_registration_id_to_designer_templates.py

```diff
@@ -0,0 +1,30 @@
+"""Add registration_form_id to designer templates
+
+Revision ID: 4e32f4d5ebe4
+Revises: e4ac92d27295
+Create Date: 2023-12-08 10:42:45.632227
+"""
+
+import sqlalchemy as sa
+from alembic import op
+
+
+# revision identifiers, used by Alembic.
+revision = '4e32f4d5ebe4'
+down_revision = 'b37cbc4bb129'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    op.add_column('designer_templates', sa.Column('registration_form_id', sa.Integer(), nullable=True), schema='indico')
+    op.create_index(None, 'designer_templates', ['registration_form_id'], schema='indico')
+    op.create_foreign_key(None, 'designer_templates', 'forms', ['registration_form_id'], ['id'],
+                          source_schema='indico', referent_schema='event_registration')
+    op.create_check_constraint('no_regform_if_category', 'designer_templates',
+                               '(category_id IS NULL) OR (registration_form_id IS NULL)', schema='indico')
+
+
+def downgrade():
```

Comment:
Should we delete custom fields from the template data when downgrading? Keeping the data around prevents the template from being modified unless you first delete all custom fields:
![image](https://github.com/indico/indico/assets/8739637/18c772a0-1749-499b-a46b-fb8f174186b4)

On the other hand if you ever want to upgrade afterwards you'll lose the data. If we keep the fields you can simply link to the right regform again and it'll work.

Reply:
let's keep it, it only breaks saving when you still have the data there, so not a big deal. downgrades are rare anyway, we just want to make sure we do not get a state where indico itself is broken or pages fail on load